src/BenchmarkDotNet.Core/Toolchains/Uap/DevicePortalApiWrapper.cs

```diff
@@ -0,0 +1,323 @@
+            public object Timestamp { get; set; }
+        }
+
+        public class RootObject
+        {
+            public int Frequency { get; set; }
+            public List<Event> Events { get; set; }
+        }
+
+        class ProgressStruct
+        {
+            public bool Success { get; set; }
+            public string Reason { get; set; }
+            public string CodeText { get; set; }
+            public int? Code { get; set; }
+        }
+
+        class PackagesStruct
+        {
+            public List<PackageStruct> InstalledPackages { get; set; }
+        }
+
+        public class PackageStruct
+        {
+            public string PackageFullName { get; set; }
+            public string PackageRelativeId { get; set; }
+            public string Name { get; set; }
+        }
+
+        private RestResponseCookie csrfToken;
+        private RestResponseCookie wmid;
+        private readonly RestClient client;
+
+        public DevicePortalApiWrapper(string csrf, string wmid, string devicePortalAddress)
+        {
+            this.csrfToken = new RestResponseCookie()
+            {
+                Name = "CSRF-Token",
+                Value = csrf
+            };
+            this.wmid = new RestResponseCookie()
+            {
+                Name = "WMID",
+                Value = wmid
+            };
+
+            this.client = new RestClient(devicePortalAddress);
+        }
+
+        public DevicePortalApiWrapper(string pin, string devicePortalAddress)
+        {
+            this.client = new RestClient(devicePortalAddress);
+            var csrfRequest = new RestRequest("authenticate.htm", Method.GET);
+            var csrfResponse = client.Execute(csrfRequest);
+            csrfToken = csrfResponse.Cookies.FirstOrDefault(x => x.Name == "CSRF-Token");
+
+            var authorizeRequest = new RestRequest("api/authorize/pair", Method.POST);
+            authorizeRequest.AddQueryParameter("pin", pin);
+            authorizeRequest.AddQueryParameter("persistent", 1.ToString());
+
+            var authorizeResponse = client.Execute(authorizeRequest);
+            wmid = authorizeResponse.Cookies.FirstOrDefault(x => x.Name == "WMID");
+
+            if (csrfToken == null || wmid == null)
+            {
+                throw new InvalidOperationException("Unable to authorize");
+            }
+        }
+
+        public PackageStruct DeployApplication(string fullPath)
+        {
+            var installRequest = new RestRequest("api/app/packagemanager/package", Method.POST);
+            installRequest.AddCookie(csrfToken.Name, csrfToken.Value);
+            installRequest.AddCookie(wmid.Name, wmid.Value);
+            installRequest.AddFile(Path.GetFileName(fullPath), fullPath);
+            installRequest.AddQueryParameter("package", Path.GetFileName(fullPath));
+            installRequest.AddHeader("X-CSRF-Token", csrfToken.Value);
+            int i = 10;
+            for (;;)
```

Comment:
I believe that `while(true)` would be easier to understand, or maybe we can simplify it further?

```cs
while (client.Execute(installRequest).StatusCode != HttpStatusCode.Accepted)
{
    if (--i == 0)
    {
         throw new InvalidOperationException("Install failed");
    }
}
```